core/java/iesi-core/src/main/java/io/metadew/iesi/connection/database/oracle/OracleDatabaseConnectionService.java

```diff
@@ -56,4 +58,13 @@ public Connection getConnection(OracleDatabaseConnection oracleDatabaseConnectio
             throw new RuntimeException(e);
         }
     }
+
+    public String refactorLimitAndOffset(String query) {
+        Pattern pattern = Pattern.compile("(?i)limit(?=\\s+[0-9]+\\s+(?i)offset+\\s+[0-9])");
```

Comment:
you can name the groups after 'limit' and 'offset' using (?<name>\d+). This way you can extract the limit value and offset value from the mather.